cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/common/tools/PathUtil.java

```diff
@@ -90,4 +92,32 @@ public static String getProject(String path) {
     public static String createWorkspaceRelativePath(IProject proj, String relativeDestinationPath) {
         return "/" + proj.getName() + "/" + relativeDestinationPath;
     }
+
+    /**
+     * Tries to return the relative project of a template. This is useful when a template is going to be
+     * relocated to a different project from the original one (the one were the input is located). Therefore
+     * the path is relative (i.e. projectname-core/../api/src/main...) and we need to get that relative
+     * project (i.e. projectname-api)
+     *
+     * @param filePath
+     *            the path of the template that maybe needs to be relocated
+     * @param project
+     *            the project containing the class used as input
+     * @return the same project if no relative path was found, and the relative project if a relative path has
+     *         been found
+     */
+    public static IProject getRelativeProjectIfNeeded(String filePath, IProject project) {
+        if (filePath.contains("/../")) {
+            String projectPath = project.getLocation().toFile().getParent().toString();
+            String projectName = projectPath.substring(projectPath.lastIndexOf(File.separator));
+
+            // We want to get the project name after "/../" which is the child project
+            int nextChildProject = filePath.indexOf("/../") + 4;
+            filePath = filePath.substring(nextChildProject, filePath.length());
+            filePath = projectName + "-" + filePath;
+
+            return ResourcesPlugin.getWorkspace().getRoot().getProject(PathUtil.getProject(filePath));
+        }
+        return project;
+    }
```

Comment:
This method seems much to specific to a certain maven project hierarchy. It could even be a totally different project out of scope of `../`.
Please have a look at https://github.com/devonfw/tools-cobigen/blob/99ba18b5bc589c04ef8623f7fbd9684c320aeca0/cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/generator/CobiGenWrapper.java#L557 This function should do what you want to do in a more generic way. Maybe you can move this function and reuse it in both locations.